internal/faustjs.org/docs/gutenberg/tutorial/create-a-custom-block.mdx

```diff
@@ -0,0 +1,314 @@
+
+```jsx {1,7,18} title="wp-templates/single.js"
+import blocks from '../wp-blocks';
+export default function Component(props) {
+...
+}
+Component.query = gql`
+...
+${blocks.MyCoolBlock.fragments.entry}
+query GetPost(
+...
+	post(id: $databaseId, idType: DATABASE_ID, asPreview: $asPreview) {
+		...
+		contentBlocks {
+			name
+			__typename
+			renderedHtml
+			id: nodeId
+			parentId
+			${blocks.MyCoolBlock.fragments.key}
+      	}
+	}
+...
+}
+```
+
+If you navigate to the page that contains this block, you should be able to inspect the properties in the console:
+
+<img
+    src="/docs/img/core-code-block-console.png"
+    alt="Code Code preview of the block attributes in the console"
+  />
+
+## 5. Implement the Block using the provided block attributes and the Editor view
+
+Let's look at the following files you generated during your custom block creation:
+
+`style.scss`: Contains the front-end block styles. You can definitely include those in the global styles section of your app, if you wish:
+
+```scss title="style.scss"
+.my-block-code {
+	// This block has customizable padding, border-box makes that more predictable.
+	box-sizing: border-box;
+
+	// Provide a minimum of overflow handling and ensure the code markup inherits
+	// the font-family set on pre.
+	code {
+		font-family: inherit;
+	}
+}
+...
+```
+
+`block.js`: Contains the block attributes specification.
+
+`save.js`: According to [the docs](https://developer.wordpress.org/block-editor/reference-guides/block-api/block-edit-save/#save),
+the `save` function defines the way in which the different attributes should be combined into the final markup, which is then serialized into `post_content`.
+
+Most of the time, the `save` function contains the actual implementation of the block, so you can take this markup and adopt it. This way, the design of the block in the editor will match the design we make in the Decoupled site.
+
+So go ahead and provide the implementation of the Block based on the `save` function. 
+
+```jsx title="wp-blocks/MyCoolBlock.js" {6}
+import { gql } from '@apollo/client';
+import getStyles from '../utilities/getStyles.js';
+
+export default function MyCoolBlock(props) {
+  const className = 'my-block-code';
+  const styles = getStyles(props.attributes)
+  return (
+    <pre className={className} style={styles}>
+      <code>{`${props.attributes?.content}`}</code>
+    </pre>
+  );
+}
+```
+
+A few important things to consider here:
+
+ - First we assign the same className `my-block-code` for the block so that we can apply the same styles as the Editor.
+ - Then we use a helper `getStyles` that creates the final inline styles for that block based on the block attributes.
+ - Finally we pass on the `props.attributes?.content` as a string to the `code` element.
+
+Let's delve into the `getStyles` helper function and see what it does:
```

Comment:
No need to explain this part so you can omit it.

Reply:
Lines 169-291 can be removed here.

Reply:
Done, thank you!